src/search/FindUtils.js

```diff
@@ -522,6 +522,28 @@ define(function (require, exports, module) {
         };
     }
 
+    /**
+     * Parses a string and replace the \r, \n and \t sequences of characters
+     * with a character based on the regex meaning of these sequences.
+     * \n => Line Feed
+     * \r => Carriage Return
+     * \t => Tab
+     */
+    function parseString(string) {
+        return string.replace(/\\(.)/g, function (match, ch) {
+            if (ch === "n") {
+                return "\n";
+            }
+            if (ch === "r") {
+                return "\r";
+            }
+            if (ch === "t") {
+                return "\t";
+            }
+            return ch;
+        });
+    }
```

Comment:
While I agree it is a bit generic, I prefer to preserve the name because it matches the one in [CodeMirror](https://github.com/codemirror/CodeMirror/blob/dbb2a5c270f2b2e626ae03157ed3863626268a6d/addon/search/search.js#L79). But it is not a big deal to change it if you really want.
To insert the chars `\` and `n` in regex mode you need to escape it: `\\n`.
I'll look at adding some tests.